.github/workflows/test.yml

```diff
@@ -0,0 +1,116 @@
+  pull_request:
+
+env:
+  FAF_DB_VERSION: v112
+  FLYWAY_VERSION: 7.5.4
+
+jobs:
+  # Static Analysis
+  isort:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+
+      - name: Install dependencies
+        run: pip install isort
+
+      - run: isort . --check --diff
+
+  # Tests
+  unit-test:
+    runs-on: ubuntu-latest
+    services:
+      faf-db:
+        image: mysql:5.7
+        ports:
+          - 3306:3306
+        options: >-
+          --health-cmd "mysqladmin ping --silent"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        env:
+          MYSQL_ROOT_PASSWORD: banana
+          MYSQL_DATABASE: faf-league
+          MYSQL_USER: faf-python-server
+          MYSQL_PASSWORD: banana
+
+      faf-rabbitmq:
+        image: rabbitmq:3-alpine
+        ports:
+          - 5672:5672
+        options: >-
+          --name faf-rabbitmq
+          --health-cmd "rabbitmq-diagnostics check_running --silent"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        env:
+          RABBITMQ_PID_FILE: /var/lib/rabbitmq/pid
+          RABBITMQ_DEFAULT_USER: admin
+          RABBITMQ_DEFAULT_PASS: banana
+          RABBITMQ_DEFAULT_VHOST: /
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.7
+
+      - name: Run db migrations
+        env:
+          FLYWAY_URL: jdbc:mysql://localhost/faf?useSSL=false
+          FLYWAY_USER: root
+          FLYWAY_PASSWORD: banana
+          FLYWAY_LOCATIONS: filesystem:db/migrations
+        run: |
+          git clone --depth 1 --branch ${FAF_DB_VERSION} https://github.com/FAForever/db
+          wget -qO- https://repo1.maven.org/maven2/org/flywaydb/flyway-commandline/${FLYWAY_VERSION}/flyway-commandline-${FLYWAY_VERSION}-linux-x64.tar.gz | tar xz
+          flyway-${FLYWAY_VERSION}/flyway migrate
+          pipenv run migrate-develop
+      - name: Setup RabbitMQ
+        run: ./.github/workflows/scripts/init-rabbitmq.sh
+
+
+      - name: Install dependencies with pipenv
+        run: |
+          pip install pipenv
+          pipenv sync --dev
+      - run: pipenv run tests --cov-report=xml -m ""
+
+      - name: Report coverage
+        uses: codecov/codecov-action@v1
+        with:
+          files: coverage.xml
+          fail_ci_if_error: true
+
+      - name: Check documentation
+        run: PYTHONWARNINGS='error::UserWarning' pipenv run pdoc3 server >/dev/null
+
+  docker-build:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+
+        - name: Build docker image
+          run: docker build --build-arg GITHUB_REF -t test_image .
+
+        - name: Test image
+          run: |
+            docker run --rm -d -p 8001:8001 test_image
```

Comment:
Probably not using this port for league service

Reply:
What ports does the league service use?

Reply:
I'm not sure. How would I check?
Also, isn't this just to test that the image works? So the port is arbitrary and just has to match to the port netcat is listening on in the next line?

Reply:
No, 8001 is the lobby server port so adding this command line option lets it bind to the host port 8001 so it can be tested from the outside. If the league service doesn’t set up a listener on tcp port 8001 then adding this command line switch will accomplish nothing.

Reply:
I think the league service just communicates through rabbitmq right? So I don’t think you need to bind any ports at all since it already has access to the rabbitmq ports over the docker network.

Reply:
Yes, it only communicates over rabbitmq. So the whole `Test image ` block can be deleted?

Reply:
That’s probably fine. The reason I added it on the lobby server was because the main file (server.py) is not covered by tests, and it had happened in the past that some change would slip through the CI where the server would crash on startup. The ‘test image’ just makes sure that the docker image that is produced actually starts. 

For league service it might be kindof hard to test that since you’d probably have to send a rabbitmq message and see if it responds. It’s probably not worth spending the time to set that up and get it working.

Reply:
Especially because the league service isn't really core infrastructure. We could live with it not being available for some hours/days.